### PR TITLE
feat: collect device info

### PR DESCRIPTION
**Pull request by BumbleFlash**
This PR collects the device information once the user registration is complete. The code is derived from OneBusAway app. 

The information collected include:-

1. App version
2. Device Model
3. SDK Version
4. SDK Version Int
5. Google Play Services App version
6. Google Play Serverice Library
7. Is Talk back Enabled or not
8. Timestamp
9. Is power saving mode enabled or not 
10. Is ignoring battery optimization or not

### CODE REVIEW

File: app/src/main/java/com/simplecity/amp_library/ShuttleApplication.java [hunk 1]
```diff
@@ -430,6 +433,20 @@ public void newUiEvent(UiEventType uiEventType){
         FirebaseIOUtils.saveUiEvent(uiEvent);
     }
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/ShuttleApplication.java:435]: Can you please add Javadocs to this method? I think they exist in the OBA version?

File: app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package edu.usf.sas.pal.muser.manager;
```
Comment (barbeau) [inline app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java:16]: Can you please add license header from original OBA file and update date to "-2021"?

File: app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import edu.usf.sas.pal.muser.constants.EventConstants;
+import edu.usf.sas.pal.muser.model.DeviceInfo;
+import edu.usf.sas.pal.muser.model.Event;
+import edu.usf.sas.pal.muser.util.FirebaseIOUtils;
+import edu.usf.sas.pal.muser.util.PreferenceUtils;
+
+import static android.content.Context.ACCESSIBILITY_SERVICE;
+
+public class DeviceInformationManager {
```
Comment (barbeau) [inline app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java:38]: Can you please add Javadoc, including mention that this is derived from OneBusAway Android class and link to that class?

File: app/src/main/java/edu/usf/sas/pal/muser/model/DeviceInfo.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package edu.usf.sas.pal.muser.model
```
Comment (barbeau) [inline app/src/main/java/edu/usf/sas/pal/muser/model/DeviceInfo.kt:1]: Same - if this is derived from OBA, could you please add license and link to that class?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 5]
```diff
@@ -21,6 +21,7 @@ object Dependencies {
         const val daggerAssistedInject = "0.3.2"
         const val expandableRecyclerView = "3.0.0-RC1"
         const val billing = "1.2"
+        const val apache = "lang3:3.0"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: I think `lang3:` should really be part of the below library name, and this variable should just be `3.0`, if `3.0` is the version number.

**Review by barbeau — COMMENTED** [review]
Thanks @BumbleFlash, looks good! A few comments in-line.

File: app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    private Context mApplicationContext;
```
Comment (barbeau) [inline app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java]: There is a general trend in Android for moving away from "Hungarian notation", or prefixing class member variables with `m`. Please refactor to `applicationContext` instead. Same for any other class member variable prefixed by `m`.

File: app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java (hunk 2 above)
Comment (BumbleFlash) [reply app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java:16]: This class was created to mimic the use of the `UpdateDeviceInfoWorker.java` https://github.com/OneBusAway/onebusaway-android/blob/master/onebusaway-android/src/main/java/org/onebusaway/android/travelbehavior/io/worker/UpdateDeviceInfoWorker.java and doesn't really exist. Should I still add a license header?

Comment (barbeau) [reply app/src/main/java/edu/usf/sas/pal/muser/manager/DeviceInformationManager.java:16]: Yes, if it's derived from OBA code and includes a large chunk of it nearly verbatim please use the license header from the class that the code came from.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 7]
```diff
@@ -21,6 +21,7 @@ object Dependencies {
         const val daggerAssistedInject = "0.3.2"
         const val expandableRecyclerView = "3.0.0-RC1"
         const val billing = "1.2"
+        const val apache = "3:3.0"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: I believe `3:` here should still be moved to the library name, as it's part of the library name. The version number is `3.0`. https://mvnrepository.com/artifact/org.apache.commons/commons-lang3

Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Ah, sorry about that!

**Review by barbeau — APPROVED** [review]
👍

**Pull request merged**